exporters/etw/include/opentelemetry/exporters/etw/etw_tracer.h

```diff
@@ -0,0 +1,1077 @@
+#include "opentelemetry/nostd/variant.h"
+
+#include "opentelemetry/common/key_value_iterable_view.h"
+
+#include "opentelemetry/trace/span.h"
+#include "opentelemetry/trace/span_context_kv_iterable_view.h"
+#include "opentelemetry/trace/span_id.h"
+#include "opentelemetry/trace/trace_id.h"
+#include "opentelemetry/trace/tracer_provider.h"
+
+#include "opentelemetry/sdk/trace/exporter.h"
+#include "opentelemetry/sdk/trace/recordable.h"
+#include "opentelemetry/sdk/trace/span_data.h"
+
+#include "opentelemetry/exporters/etw/etw_fields.h"
+#include "opentelemetry/exporters/etw/etw_properties.h"
+#include "opentelemetry/exporters/etw/etw_provider.h"
+#include "opentelemetry/exporters/etw/utils.h"
+
+namespace core  = opentelemetry::core;
+namespace trace = opentelemetry::trace;
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace exporter
+{
+namespace ETW
+{
+
+/**
+ * @brief TracerProvider Options passed via SDK API.
+ */
+using TracerProviderOptions =
+    std::map<std::string, nostd::variant<std::string, uint64_t, float, bool>>;
+
+/**
+ * @brief TracerProvider runtime configuration class. Internal representation
+ * of TracerProviderOptions used by various components of SDK.
+ */
+typedef struct
+{
+  bool enableTraceId;            // Set `TraceId` on ETW events
+  bool enableSpanId;             // Set `SpanId` on ETW events
+  bool enableActivityId;         // Assign `SpanId` to `ActivityId`
+  bool enableActivityTracking;   // Emit TraceLogging events for Span/Start and Span/Stop
+  bool enableRelatedActivityId;  // Assign parent `SpanId` to `RelatedActivityId`
+  bool enableAutoParent;         // Start new spans as children of current active span
+} TracerProviderConfiguration;
+
+/**
+ * @brief Helper template to convert a variant value from TracerProviderOptions to
+ * TracerProviderConfiguration
+ *
+ * @param options           TracerProviderOptions passed on API surface
+ * @param key               Option name
+ * @param value             Reference to destination value
+ * @param defaultValue      Default value if option is not supplied
+ */
+template <typename T>
+static inline void GetOption(const TracerProviderOptions &options,
+                             const char *key,
+                             T &value,
+                             T defaultValue)
+{
+  auto it = options.find(key);
+  if (it != options.end())
+  {
+    auto val = it->second;
+    value    = nostd::get<T>(val);
+  }
+  else
+  {
+    value = defaultValue;
+  }
+}
+
+class Span;
+
+/**
+ * @brief Template that allows to instantiate new Span object for header-only forward-declared
+ * ETW::Span type
+ *
+ * @tparam SpanType     Expected to be ETW::Span
+ * @tparam TracerType   expected to be ETW::Tracer
+ * @param objPtr        Pointer to parent
+ * @param name          Span Name
+ * @param options       Span Options
+ * @return              Span instance
+ */
+template <class SpanType, class TracerType>
+SpanType *new_span(TracerType *objPtr,
```

Comment:
Probably this should be in an anonymous namespace? I guess it's not intended to be used anywhere outside this file.

Reply:
Could be potentially used if someone would like to inherit from the `etw::Tracer` class. It is in `exporter::etw` namespace only.